core/src/main/scala/slog/LogBuilder.scala

```diff
@@ -0,0 +1,7 @@
+package slog
+
+trait LogBuilder[F[_]] {
+  def msg(msg: String): F[Unit]
+  def msg(ex: Throwable)(msg: String): F[Unit]
```

Comment:
Could it possibly go as far as having separate `msg` and `ex` methods, both aliases to `context` with fixed keys?
```
def msg(msg: => String): LogBuilder[F] = context("message", msg)
def ex(ex: => Throwable): LogBuilder[F] = context("exception", Throwable)
```
Logging itself would not be executed by `msg` method, but with loglevel-determining method e.g. `debug()` or `info()` like @sideeffffect proposed earlier.

Reply:
I actually had something like this. The problem with it is that it adds at least one more allocation (temporary) object which might be costly.

Reply:
I started to like the fluent/builder interface, but maybe we *should* consider using plain methods ala `logger.info("some string", o, p, ex)`

Reply:
Those methods are very difficult to read. At least for me.

Reply:
but less allocations.. I don't really know the cost and don't care that much myself :man_shrugging:

Reply:
Do we really care that much about extra allocations? I understand the concern especially for logging library, but to let it guide API decisions is quite extreme.

Also if we decide to use GraalVM and I understood correctly the magic behind it, these allocations should be eliminated (from heap).

Reply:
graal won't help you here, most people who use this (even us) won't run this on graal jdk